ccloader/js/modset.js

```diff
@@ -0,0 +1,87 @@
+
+function assert(bool, message = '') {
+	if (!bool) {
+		throw message;
+	}
+}
+export class Modset {
+	constructor(modloader, manifestPath) {
+		this.modloader = modloader;
+		this.manifestPath = manifestPath;
+
+		this.manifest = null;
+		this.name = "";
+		this.mods = [];
+
+		this.baseDirectory = this._getBaseName(manifestPath).replace(/\\/g, '/').replace(/\/\//g, '/') + '/';
+
+		this.loaded = false;
+
+	}
+
+	async load() {
+		try {
+			const manifest = await this._parse(this.manifestPath);
+			this.manifest = manifest;
+
+			const {name, mods, searchPaths} = manifest;
```

Comment:
There should also be a display name. For example, `'default'` should be displayed as `Default`

Reply:
You might as well add `description` and `icon`